### PR TITLE
client improvements

### PR DESCRIPTION
**Pull request by LawlAoux**

### CODE REVIEW

File: config_to_bot/debugger/ui.js [hunk 1]
```diff
@@ -7,7 +7,19 @@ import WebSocket from "ws";
 const textToBotUtterance = ({ botUtterance, state }) => (
   <Box>
     <Text color="white">🤖 {botUtterance}</Text>
-    {state ? <Text color="red"> {JSON.stringify(state)}</Text> : null}
+    {state != null & state != [] ? (
+      <Text color="red">
+        {" "}
```
Comment (uriva) [inline config_to_bot/debugger/ui.js:12]: consider using margin instead of this space

File: config_to_bot/debugger/ui.js [hunk 2]
```diff
@@ -7,7 +7,19 @@ import WebSocket from "ws";
 const textToBotUtterance = ({ botUtterance, state }) => (
   <Box>
     <Text color="white">🤖 {botUtterance}</Text>
-    {state ? <Text color="red"> {JSON.stringify(state)}</Text> : null}
+    {state != null & state != [] ? (
```
Comment (uriva) [inline config_to_bot/debugger/ui.js:10]: I think `!= []` is always `true` in js (pointer comparison, not value comparison)

File: config_to_bot/debugger/ui.js [hunk 3]
```diff
@@ -7,7 +7,19 @@ import WebSocket from "ws";
...
+      <Text color="red">
+        {" "}
+        {JSON.stringify(
+          state.map((element) => {
+            if (element == null) {
```
Comment (uriva) [inline config_to_bot/debugger/ui.js:15]: `(element) => element == null ? "unknown" : element`

File: config_to_bot/debugger/ui.js [hunk 4]
```diff
@@ -7,7 +7,19 @@ import WebSocket from "ws";
 const textToBotUtterance = ({ botUtterance, state }) => (
   <Box>
     <Text color="white">🤖 {botUtterance}</Text>
-    {state ? <Text color="red"> {JSON.stringify(state)}</Text> : null}
+    {state != null & state != [] ? (
+      <Text color="red">
+        {" "}
+        {JSON.stringify(
```
Comment (uriva) [inline config_to_bot/debugger/ui.js:13]: might be better to map them to elements than to use `stringify`

File: config_to_bot/debugger/ui.js [hunk 5]
```diff
@@ -43,10 +55,15 @@ const FullScreen = ({ children }) => {
 };
 
 const App = () => {
-  const [events, addEvent] = useReducer(
-    (state, current) => [...state, current],
-    []
-  );
+  const [events, addEvent] = useReducer((state, current) => {
+    if (
+      current.userUtterance == "reset" ||
```
Comment (uriva) [inline config_to_bot/debugger/ui.js:60]: `[x, y].includes(z)`

File: config_to_bot/debugger/ui.js [hunk 6]
```diff
@@ -43,10 +55,15 @@ const FullScreen = ({ children }) => {
...
+      current.userUtterance == "reload"
+    ) {
+      return [];
```
Comment (uriva) [inline config_to_bot/debugger/ui.js:63]: use a ternary and use an implicit return (`=> x`)

**Review by uriva — APPROVED** [review]

File: config_to_bot/debugger/ui.js (hunk 4 above)
Comment (LawlAoux) [reply config_to_bot/debugger/ui.js]: Not sure what you mean here, now that I have an object I just stringify it

File: config_to_bot/debugger/ui.js (hunk 5 above)
Comment (LawlAoux) [reply config_to_bot/debugger/ui.js]: It doesn't work.. I think because it is the opposite direction

File: config_to_bot/debugger/ui.js (hunk 4 above)
Comment (uriva) [reply config_to_bot/debugger/ui.js]: if you just stringify it you can't separate the keys from the values

File: config_to_bot/debugger/ui.js (hunk 5 above)
Comment (uriva) [reply config_to_bot/debugger/ui.js]: `["reset", "reload"].includes(current.userUtterance`

**Pull request merged**